lib/puppet/provider/fullchain/posix.rb

```diff
@@ -0,0 +1,62 @@
+require 'pp'
+require 'common'
+require 'openssl'
+
+Puppet::Type.type(:fullchain).provide(:posix) do
+  confine feature: :posix
+
+  def exists?
+    return false unless Pathname.new(resource[:path]).exist?
+
+    debug 'File found, loading contents.'
```

Comment:
You might want to prefix this kind of message. I'm not sure it will be clear in the logs that it is associated with this type.

Reply:
Puppet does wrap it up. This is pretty similar how I made it [here](https://github.com/rtib/puppet-openssl/blob/539fdd2f48ab369cdc82364dba3bcba1ab1528b6/lib/puppet/provider/cert_file/posix.rb#L10), which does an output like
```
Jul 27 13:30:30 trepasi-dev002 puppet-agent[14486]: (Cert_file[/etc/puppetlabs/puppet/ssl/certs/ca.pem](provider=posix)) Checking file format /etc/puppetlabs/puppet/ssl/certs/ca.pem
```